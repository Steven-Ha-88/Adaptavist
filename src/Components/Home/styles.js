import styled from "styled-components";
import PatternBackground from "./../../images/pattern.svg";

export const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  text-align: center;
  background-image: url(${PatternBackground});
  height: 100vh;
  background-size: cover;
`;

export const Title = styled.h1`
  font-family: "Roboto", sans-serif;
  font-weight: 100;
  color: white;
  font-size: 38px;
`;
