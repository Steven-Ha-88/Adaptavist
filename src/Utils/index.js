import {
  SVG1d,
  SVG1n,
  SVG2d,
  SVG2n,
  SVG3d,
  SVG3n,
  SVG4d,
  SVG4n,
  SVG9d,
  SVG9n,
  SVG10d,
  SVG10n,
  SVG11d,
  SVG11n,
  SVG13d,
  SVG13n,
  SVG50d,
  SVG50n
} from "../images/index";

export const findTestAttr = (component, attribute) => {
  const wrapper = component.find(`[data-test="${attribute}"]`);
  return wrapper;
};

export const toCelcius = kel => {
  const temp = Math.round(kel - 273.15);
  return temp;
};

export const weatherImg = term => {
  switch (term) {
    case "01d":
      return SVG1d;
    case "01n":
      return SVG1n;
    case "02d":
      return SVG2d;
    case "02n":
      return SVG2n;
    case "03d":
      return SVG3d;
    case "03n":
      return SVG3n;
    case "04d":
      return SVG4d;
    case "04n":
      return SVG4n;
    case "09d":
      return SVG9d;
    case "09n":
      return SVG9n;
    case "10d":
      return SVG10d;
    case "10n":
      return SVG10n;
    case "11d":
      return SVG11d;
    case "11n":
      return SVG11n;
    case "13d":
      return SVG13d;
    case "13n":
      return SVG13n;
    case "50d":
      return SVG50d;
    case "50n":
      return SVG50n;
    default:
      return null;
  }
};
