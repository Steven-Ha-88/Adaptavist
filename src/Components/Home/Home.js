import React from "react";
import SearchBar from "./../SearchBar/SearchBar";
import { Title, Container } from "./styles";

const Home = () => {
  return (
    <Container>
      <Title>Enter a City and State</Title>
      <SearchBar />
    </Container>
  );
};

export default Home;
