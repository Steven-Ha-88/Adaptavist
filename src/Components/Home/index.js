import React from "react";
import SearchBar from "../SearchBar";
import { Title, Container } from "./styles";

const Home = () => {
  return (
    <Container data-test="HomeComponent">
      <Title data-test="title">Enter a City and State</Title>
      <SearchBar data-test="search" />
    </Container>
  );
};

export default Home;
