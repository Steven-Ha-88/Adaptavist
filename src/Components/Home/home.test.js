import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";
import { findTestAttr } from "./../../Utils";
import toJson from "enzyme-to-json";

const setUp = (props = {}) => {
  const component = shallow(<Home {...props} />);
  return component;
};

describe("Home Component", () => {
  describe("Have no props", () => {
    let component;
    beforeEach(() => {
      return (component = setUp());
    });

    it("mounts", () => {
      expect(component.exists()).toEqual(true);
    });

    it("Should render without errors", () => {
      const wrapper = findTestAttr(component, "HomeComponent");
      expect(wrapper.length).toBe(1);
    });

    it("Should render title", () => {
      const wrapper = findTestAttr(component, "title");
      expect(wrapper.length).toBe(1);
    });

    it("Should render search", () => {
      const wrapper = findTestAttr(component, "search");
      expect(wrapper.length).toBe(1);
    });

    it("Should match snapshot", () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
