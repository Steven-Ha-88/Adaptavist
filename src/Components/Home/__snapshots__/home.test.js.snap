// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Have no props Should match snapshot 1`] = `
<styled.div
  data-test="HomeComponent"
>
  <styled.h1
    data-test="title"
  >
    Enter a City and State
  </styled.h1>
  <SearchBar
    data-test="search"
  />
</styled.div>
`;
